bionic/deriver.py

```diff
@@ -542,23 +553,21 @@ def __init__(self, task):
         self.case_key = None
         self.provider = None
 
-        # These are set by
-        # EntityDeriver._set_up_provenance_for_task_state(), just
-        # before the task state becomes eligible for computation.
+        # These are set by EntityDeriver._complete_task_state(), just
+        # before the task state becomes eligible for cache lookup / computation.
         self.provenance = None
         self.queries = None
-
-        # These are set by
-        # EntityDeriver._set_up_cache_accessors_for_task_state(), just before
-        # we attempt to find cached values for these tasks.
         self.cache_accessors = None
-        self.has_cached_values = None
 
         # This can be set by
-        # EntityDeriver._attempt_to_load_task_state_results() or
-        # EntityDeriver._compute_task_state().
+        # EntityDeriver._get_results_for_complete_task_state().
```

Comment:
This can also be set by `_compute_task_state`, right?

Maybe we should also make it clear that this should only be accessed via `_get_results_for_complete_task_state()`, and not directly?

Reply:
ah this can totally be set by `_compute_task_state`. I'll change the comments for both results and hashed results by name to include more info (like what you said for results by name and that hashed results by name will always be present for completed tasks if it is persisted).

Reply:
Wdyt about renaming the variable to `_` `results_by_name` so that it's clear that no one should access it directly?

Reply:
I like that idea!